torch/lib/c10d/TCPStore.cpp

```diff
@@ -220,6 +409,8 @@ void TCPStoreDaemon::deleteHandler(int socket) {
   std::string key = tcputil::recvString(socket);
   auto numDeleted = tcpStore_.erase(key);
   tcputil::sendValue<int64_t>(socket, numDeleted);
+  // Remove all clients watching the key
+  watchedSockets_.erase(key);
```

Comment:
Is this behavior correct? What if someone sets the same key after the delete? Shouldn't the user get notified of that?